hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableFactory.java

```diff
@@ -75,7 +76,7 @@ public DynamicTableSource createDynamicTableSource(Context context) {
     Path path = new Path(conf.getOptional(FlinkOptions.PATH).orElseThrow(() ->
         new ValidationException("Option [path] should not be empty.")));
     setupTableOptions(conf.getString(FlinkOptions.PATH), conf);
-    ResolvedSchema schema = context.getCatalogTable().getResolvedSchema();
+    ResolvedSchema schema = ResolvedSchema.of(context.getCatalogTable().getResolvedSchema().getColumns().stream().filter(c-> !(c instanceof Column.ComputedColumn)).collect(Collectors.toList()));
     setupConfOptions(conf, context.getObjectIdentifier(), context.getCatalogTable(), schema);
```

Comment:
in `ITTestHoodieDataSource.testWriteReadWithComputedColumns` we add a test case for the computed column and it works correctly, can you explain a little more why the `proc_time` column failes instead?

Reply:
org.apache.hudi.table.format.cow.ParquetSplitReaderUtil#genPartColumnarRowReader
```
    List<String> selNonPartNames = Arrays.stream(selectedFields)
        .mapToObj(i -> fullFieldNames[i])
        .filter(n -> !partitionSpec.containsKey(n))
        .collect(Collectors.toList());
```
selectFields=[0,1,2,3]，fullFieldNames=[id,name,create_date]，fullFieldNames is from  commited metadata.

Reply:
So the proc_time does not write into hudi at all, can you write a test case for it, just like we do in ITTestHoodieDataSource.testWriteReadWithComputedColumns

Reply:
In `createDynamicTableSink` we also use the `context.getCatalogTable().getResolvedSchema()`, curious why it works though

Reply:
`org.apache.flink.table.catalog.ResolvedSchema#toSinkRowDataType` will filter ComputedColumn whitch is not persisted, so computed column never write into hudi.

```
    public DataType toSinkRowDataType() {
        final DataTypes.Field[] fields =
                columns.stream()
                        .filter(Column::isPersisted)
                        .map(ResolvedSchema::columnToField)
                        .toArray(DataTypes.Field[]::new);
        return ROW(fields).notNull();
    }
```